noxfiles/test_docker_nox.py

```diff
@@ -60,13 +68,15 @@ def test_different_path(self) -> None:
         )
         assert actual_result == expected_result
 
+
+class TestGetBuildxCommands:
```

Comment:
@adamsachs I changed up the tests here and would like your thoughts!

I made a major change away from the mock-based tests for a few reasons:
1. I am honestly biased against mocking for the most part. I think mocking is a bit of a crutch to avoid rewriting code in a more testable manner (weakly held strong belief). Not saying that's why you wrote mock-based tests, just saying why I shy away from it
2. It tested more of the execution in my opinion instead of focus testing the command/generation output
3. Perhaps most importantly, mocking gets complicated, and the things I would've had to mock to get it to "fake" execute given the new complexity of multiprocessing didn't feel right. Simpler and more concise tests felt _better_ to me, subjectively in this case

All that being said, I also don't want to inadvertently leave any testing gaps open. Please check me here if you think I've missed out on testing functionality and I've regressed our test coverage